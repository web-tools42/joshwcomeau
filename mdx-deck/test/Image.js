import React from 'react'
import { create as render } from 'react-test-renderer'
import 'jest-styled-components'
import Image from '../src/Image'

describe('Image', () => {
  test('renders', () => {
    const json = render(<Image />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with image', () => {
    const json = render(<Image src="kitten.png" />).toJSON()
    expect(json).toHaveStyleRule('background-image', 'url(kitten.png)')
  })

  test('renders with custom css', () => {
    const json = render(<Image css={{ backgroundColor: 'tomato' }} />).toJSON()
    expect(json).toHaveStyleRule('background-color', 'tomato')
  })
})
