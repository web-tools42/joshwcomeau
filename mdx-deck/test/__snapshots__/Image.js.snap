// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders 1`] = `
.c0 {
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-image: url(undefined);
  width: 100vw;
  height: 100vh;
}

<div
  className="c0"
  height="100vh"
  width="100vw"
/>
`;
